scripts/execute_task_for_changed_modules.sh

```diff
@@ -3,29 +3,55 @@ echo -----Fetching orign master
 git fetch origin master:refs/remotes/origin/master
 echo -----Done fetching orign master
 
-# directory names that corresponds to a module that contains the task - this list needs to be manually updated when a new module is added/deleted
-TESTABLE_MODULES="payments payments-core payments-model paymentsheet wechatpay link stripecardscan identity financial-connections stripe-core payments-ui-core camera-core stripe-ui-core payment-method-messaging"
+# directory names that corresponds to all modules, space delimited.
+TESTABLE_MODULES=`./gradlew projects |  grep -E ".*--- Project ':.*'$" | sed -r "s/^.*--- Project ':(.+)'$/\1/" | tr '\n' ' '`
+
+echo "Checking Modules: ${TESTABLE_MODULES}"
+
 # a function to check if a dir is in TESTABLE_MODULES
 isTestableModule() {
   [[ $TESTABLE_MODULES =~ (^|[[:space:]])$1($|[[:space:]]) ]]
 }
 
+# Files or directories that trigger all modules to run if changed
+CRITICAL_DEPS="build.gradle settings.gradle dependencies.gradle"
+# a function to check if a file/dir is in CRITICAL_DEPS
+isCriticalDeps() {
+  [[ $CRITICAL_DEPS =~ (^|[[:space:]])$1($|[[:space:]]) ]]
+}
+
 # Determines the list passed in as $1 contains the element passed in as $2
 listContainsElement() {
   [[ $1 =~ (^|[[:space:]])$2($|[[:space:]]) ]]
 }
 
+# Determines if the module passed in as $1 contains the task passed in as $2
+projectContainsTask() {
+  module=$1
+  task=$2
+  result="$(./gradlew :${module}:tasks | grep -E ^${task} | wc -l)"
+  [[ result -ge 1 ]]
+}
+
+tasks_to_run=$@
+
 # find all dirs changed through git diff
 changed_dirs=""
 while read line; do
+  if isCriticalDeps $line; then
+    echo "critical dep $line is changed, running task (${tasks_to_run}) for all modules."
+    echo "./gradlew ${tasks_to_run}"
+    eval "./gradlew ${tasks_to_run}"
+    exit
+  fi
+
   module_name=${line%%/*} # This gets the first word before '/'
   # add this dir if we haven't add it yet
   if ! listContainsElement "${changed_dirs[@]}" $module_name; then
     changed_dirs="$changed_dirs $module_name" # string concat
   fi
 done < <(git diff --name-only remotes/origin/master)
```

Comment:
I was testing this by commenting out this line, and adding a file like this command would generate.
```suggestion
done < <(cat example.txt)
```

Example.txt file:
```
README.md
VERSION
financial-connections/build.gradle
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_0.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_1.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_10.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_11.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_12.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_13.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_14.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_15.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_16.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_17.png
financial-connections/screenshots/debug/com.stripe.android.financialconnections.screenshottests.FinancialConnectionsShotShowkaseScreenshotTest_ShowkaseCodegen_composable_screenshot_test_18.png
```

I'd also comment out the `eval ....` lines, so I could see the command it generated, without running the actual commands, since that would take forever.